### PR TITLE
[CARBONDATA-3947]Fixed Hive read/write operation for Insert into Select operation.

### PR DESCRIPTION
**Pull request by nihal0107**
### Why is this PR needed?
 FileNotFound Exception is thrown for Hive read/write operation.
 
 ### What changes were proposed in this PR?
When carbon schema file is not present at given location then build the table from the configuration at place of throwing exception.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - Yes

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1876/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3615/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3668/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1929/

File: integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonInputFormat.java [hunk 1]
```diff
@@ -213,6 +223,45 @@ private QueryModel getQueryModel(Configuration configuration, String path)
         .build();
   }
 
+  private static CarbonTable getCarbonTable(Configuration tableProperties)
```
Comment (kunal642) [inline integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonInputFormat.java:226]: Please move this method to HiveCarbonUtil class.

File: integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonInputFormat.java [hunk 2]
```diff
@@ -213,6 +223,45 @@ private QueryModel getQueryModel(Configuration configuration, String path)
...
+
+    AbsoluteTableIdentifier absoluteTableIdentifier = AbsoluteTableIdentifier
+        .from(tableProperties.get(hive_metastoreConstants.META_TABLE_LOCATION),
+            getDatabaseName(tableProperties), getTableName(tableProperties));
+    String schemaPath =
+        CarbonTablePath.getSchemaFilePath(absoluteTableIdentifier.getTablePath(), tableProperties);
+
+    CarbonTable carbonTable;
+    String carbonDataFile = CarbonUtil.getFilePathExternalFilePath(schemaPath, tableProperties);
+    if (carbonDataFile == null) {
```
Comment (kunal642) [inline integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonInputFormat.java:253]: This method is called only when carbondataFile does not exist. Refer: https://github.com/apache/carbondata/pull/3878/files#diff-e4da0735e7ef96dca3a7bfedb5a8039dR96

Please remove this uncessary listing and directly use the CarbonTable.buildFromTableInfo() to build carbon table.

No need for inferSchema code also

File: integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonInputFormat.java [hunk 3]
```diff
@@ -202,7 +212,7 @@ protected void setFilterPredicates(Configuration configuration, CarbonTable carb
   }
 
   private QueryModel getQueryModel(Configuration configuration, String path)
-      throws IOException, InvalidConfigurationException {
+      throws IOException, InvalidConfigurationException, SQLException {
```
Comment (kunal642) [inline integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonInputFormat.java:215]: Which method is throwing SQLException?

File: integration/hive/src/main/java/org/apache/carbondata/hive/util/HiveCarbonUtil.java [hunk 4]
```diff
@@ -155,7 +155,7 @@ public static CarbonLoadModel getCarbonLoadModel(String tableName, String databa
     return loadModel;
   }
 
-  private static TableInfo getTableInfo(String tableName, String databaseName, String location,
+  public static TableInfo getTableInfo(String tableName, String databaseName, String location,
```
Comment (kunal642) [inline integration/hive/src/main/java/org/apache/carbondata/hive/util/HiveCarbonUtil.java:158]: revert this change after moving getCarbonTable to HiveCarbonUtil

File: integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonInputFormat.java (hunk 3 above)
Comment (nihal0107) [reply integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonInputFormat.java:215]: getTableInfo() is throwing SQLException which is called from getCarbonTable().

File: integration/hive/src/main/java/org/apache/carbondata/hive/util/HiveCarbonUtil.java (hunk 4 above)
Comment (nihal0107) [reply integration/hive/src/main/java/org/apache/carbondata/hive/util/HiveCarbonUtil.java]: done

File: integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonInputFormat.java (hunk 2 above)
Comment (nihal0107) [reply integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonInputFormat.java]: done.

File: integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonInputFormat.java (hunk 1 above)
Comment (nihal0107) [reply integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonInputFormat.java]: done

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3693/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1954/

**Comment by kunal642** [comment]
LGTM
